File: Beginner/Chef and Rainbow Array (RAINBOWA)/rainbow.c
#include <stdio.h>

int main(){
	int t;
	scanf("%d", &t);
	
	while (t--){
		
		int N;
		scanf("%d", &N);
		
		int arr[N], i, j, k;
		int flag = 0;
		
		for (i=0; i<N; i++)
			scanf("%d", &arr[i]);
		
		
		for (j=0, k=N-1; j<=N/2; j++,k--){
			if (arr[j] > flag+1)
				break;
			flag = arr[j];

			if (arr[j] != arr[k])
				break;
		}	
		
		if (flag==7 && j==N/2+1)
			printf("yes\n");
		else
			printf("no\n");
	
	}
	
	return 0;
}

File: Beginner/Chef and Rainbow Array (RAINBOWA)/rainbow.py
t = int(input())


while t:
    n = int(input())
    a = list(map(int,input().split()))
    r = {1,2,3,4,5,6,7}                        
    d = a[::-1]      
    if a == d and set(a) == r:
        print('yes')
    else:
        print('no')

    t = t-1